lredis/commands.lua

```diff
@@ -1,79 +1,150 @@
-local methods = {}
+local base_methods = {}
 
-function methods:call(...)
+-- client methods and metatable
+local client_methods = setmetatable({}, { __index = base_methods })
```

Comment:
I'm not a huge fan of inheritance like this.

Better to just keep them on the client object.

Reply:
I disagree.  If we keep everything on the client object then we have to make much larger code changes to guard against incorrect use.  This way we let Lua do the checking for us.